python 3.7.3 less version will fail for task type changing

latest dev branch.
I noticed that some bamboo jobs failed here:
http://idm-bamboo:8085/browse/RIT-WIDMT37-149
For example for test_tasks under idmtools_model tests.(and other 3 cases)
```
collection failure
test_tasks.py:5: in <module>
    from idmtools.entities.command_task import CommandTask
..\idmtools\entities\command_task.py:3: in <module>
    from idmtools.entities.itask import ITask
..\idmtools\entities\itask.py:22: in <module>
    TTaskHook = Callable[[TTaskParent], NoReturn]
C:\Python37\Lib\typing.py:755: in __getitem__
    return self.__getitem_inner__(params)
C:\Python37\Lib\typing.py:251: in inner
    return func(*args, **kwds)
C:\Python37\Lib\typing.py:774: in __getitem_inner__
    result = _type_check(result, msg)
C:\Python37\Lib\typing.py:135: in _type_check
    raise TypeError(f"Plain {arg} is not valid as type argument")
E   TypeError: Plain typing.NoReturn is not valid as type argument

```
Bamboo agent for that box was python 3.7.0 while it was running.
this is due to typings.py is different between python 3.7.0 and python 3.7.3+ in 
def _type_check(arg, msg, is_argument=True):
For 3.7.0
```
    if (isinstance(arg, _SpecialForm) and arg is not Any or
            arg in (Generic, _Protocol)):
        raise TypeError(f"Plain {arg} is not valid as type argument")
```
For 3.7.3, same line changed to:
   ```
 if (isinstance(arg, _SpecialForm) and arg not in (Any, NoReturn) or
            arg in (Generic, _Protocol)):
        raise TypeError(f"Plain {arg} is not valid as type argument")

```
After I upgrade to python 3.7.6 for that windows bamboo agent, everything passed.

So we may need to document python version to 3.7.3+ for idmtools.

python 3.6 seems OK. not type hint check

BTW, python3.7.0 failed this line in simulation.py:
    `pre_creation_hooks: List[Callable[[], NoReturn]] = field(default_factory=lambda: [Simulation.gather_assets])`

--- a/idmtools_core/setup.py
+++ b/idmtools_core/setup.py
@@ -19,6 +19,8 @@ test_requirements = ['pytest', 'pytest-runner', 'numpy==1.16.4', 'xmlrunner', 'p
 # check for python 3.6
 if sys.version_info[1] == 6:
     requirements.append('dataclasses')
+elif sys.version_info[0] == 3 and sys.version_info[1] == 7 and sys.version_info[2] < 3:
+    raise EnvironmentError("Python 3.7 requires 3.7.3 or higher")
 
 extras = {
     'test': test_requirements,

--- a/idmtools_models/setup.py
+++ b/idmtools_models/setup.py
@@ -2,6 +2,7 @@
 # -*- coding: utf-8 -*-
 
 """The setup script for the idmtools_models platform, core tools for other models such as Python and R models."""
+import sys
 
 from setuptools import setup, find_packages
 
@@ -16,6 +17,10 @@ setup_requirements = []
 test_requirements = ['pytest', 'pytest-runner', 'pytest-timeout', 'pytest-cache'] + build_requirements
 
 extras = dict(test=test_requirements, packaging=build_requirements)
+
+# check for python 3.7
+if sys.version_info[0] == 3 and sys.version_info[1] == 7 and sys.version_info[2] < 3:
+    raise EnvironmentError("Python 3.7 requires 3.7.3 or higher")
 
 authors = [
     ("Ross Carter", "rcarter@idmod.org"),

